data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/accumulator/InMemoryBuffer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.accumulator;
+
+import org.apache.commons.lang3.time.StopWatch;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A buffer can hold in memory data and flushing it to S3.
+ */
+public class InMemoryBuffer implements Buffer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(InMemoryBuffer.class);
+    private final ByteArrayOutputStream byteArrayOutputStream;
+    private int eventCount;
+    private final StopWatch watch;
+
+    InMemoryBuffer() {
+        byteArrayOutputStream = new ByteArrayOutputStream();
```

Comment:
Can we reset the stream instead of creating a new stream object every time?

Reply:
Modified the code to reset the stream instead of creating a new stream object every time.